aws/templates/id/id_apigateway.ftl

```diff
@@ -185,30 +160,71 @@
-                "apigateway" : {
-                    "Id" : id,
-                    "Name" : name
+                APIGATEWAY_RESOURCE_TYPE : {
+                    "Id" : apiId,
+                    "Name" : apiName
+                },
+                APIGATEWAY_DEPLOY_RESOURCE_TYPE : {
+                    "Id" : formatResourceId(APIGATEWAY_DEPLOY_RESOURCE_TYPE, core.Id, runId)
+                },
+                APIGATEWAY_STAGE_RESOURCE_TYPE : {
+                    "Id" : stageId,
+                    "Name" : core.Version.Name
+                },
+                APIGATEWAY_DOMAIN_RESOURCE_TYPE : {
+                    "Id" : formatDependentResourceId(APIGATEWAY_DOMAIN_RESOURCE_TYPE, apiId)
+                },
+                APIGATEWAY_BASEPATHMAPPING_RESOURCE_TYPE : { 
+                    "Id" : formatDependentResourceId(APIGATEWAY_BASEPATHMAPPING_RESOURCE_TYPE, stageId)
+                },
+                APIGATEWAY_USAGEPLAN_RESOURCE_TYPE : {
+                    "Id" : formatDependentResourceId(APIGATEWAY_USAGEPLAN_RESOURCE_TYPE, cfId),
+                    "Name" : formatComponentFullName(core.Tier, core.Component, occurrence)
+                },
+                "invalidlogmetric" : {
+                    "Id" : formatDependentLogMetricId(stageId, "invalid"),
+                    "Name" : "Invalid"
+                },
+                "invalidalarm" : {
+                    "Id" : formatDependentAlarmId(stageId, "invalid"),
+                    "Name" : formatComponentAlarmName(core.Tier, core.Component, occurrence,"invalid")
+                },
+                "cf" : {
+                    "Id" : cfId,
+                    "Name" : formatComponentCFDistributionName(core.Tier, core.Component, occurrence)
+                },
+                "cforigin" : {
+                    "Id" : "apigateway"
+                },
+                "wafacl" : { 
+                    "Id" : formatDependentWAFAclId(apiId),
+                    "Name" : formatComponentWAFAclName(core.Tier, core.Component, occurrence)
+                },
+                "docs" : {
+                    "Id" : formatS3Id(core.Id, "docs")
+                },
+                "docspolicy" : {
+                    "Id" : formatBucketPolicyId(core.Id, "docs")
                 }
             },
             "Attributes" : {
                 "FQDN" : fqdn,
-                "URL" : "https://" + fqdn + versionPath,
-                "DOCS_URL" : "http://docs." + fqdn,
+                "URL" : "https://" + fqdn,
                 "SIGNING_FQDN" : signingFqdn,
                 "SIGNING_URL" : "https://" + signingFqdn + versionPath,
                 "INTERNAL_FQDN" : internalFqdn,
-                "INTERNAL_URL" : "https://" + internalFqdn + versionPath,
-                "INTERNAL_PATH" : internalPath
+                "INTERNAL_URL" : "https://" + internalFqdn,
+                "DOCS_URL" : "http://" + getExistingReference(docsId, NAME_ATTRIBUTE_TYPE)
```

Comment:
NAME_ATTRIBUTE looks wrong - should be a URL attribute?

Reply:
The S3 website url contains the aws suffix on the end - ap-southeast-2.aws.com or something like that. The name is actually the FQDN that we want to use for the Docs site. Since thats the DNS entry we put in.